src/main/java/bio/terra/cli/command/resource/Resolve.java

```diff
@@ -60,6 +61,17 @@ protected void execute() {
       default:
         cloudId = resource.resolve();
     }
-    formatOption.printReturnValue(cloudId);
+    JSONObject object = new JSONObject();
+    object.put(resource.getName(), cloudId);
+    formatOption.printReturnValue(object, this::printText, this::printJson);
+  }
+
+  private void printText(JSONObject object) {
+    Resource resource = Context.requireWorkspace().getResource(resourceNameOption.name);
+    OUT.println(object.get(resource.getName()));
+  }
+
+  private void printJson(JSONObject object) {
+    OUT.println(object.toString());
```

Comment:
I don't think you need an additional signature to do custom printJson + default printText, you could use the method specifying both but use the Format::printText since it's public and static (something like `formatOption.printReturnValue(object, Format::printText, this::printJsonValue);`).

It only saves one duplicate line in a method that's probably not going to change though, so either way is fine